pkg/route/default.go

```diff
@@ -140,6 +141,7 @@ type defaultRoute struct {
 	method   string
 	parts    []Part
 	ctx      *routeMatchContext
+	middleware []middleware.Middleware
```

Comment:
It's generally OK for values inside of structs, but remember to be careful with variable names that match package names. No change needed here.

Reply:
Based on Go's rules regarding scoping and shadowing, I'm pretty sure this will never result in ambiguity. Afaik it's pretty common to name properties with their type name (when semantically valid to do so), as the std lib does in the screenshot below. Since `middleware` is private, I lowercased
![image](https://user-images.githubusercontent.com/35514810/228927582-5507b21c-6e08-45f2-ba12-d6d5655c4d69.png)

Reply:
Yeah, this was more of a personal nitpick than an expectation that something breaks or is wrong--thanks for the great clarification though!